src/hooks/useSearchResults.ts

```diff
@@ -45,13 +48,20 @@ export const useSearchResults = () => {
             slug: `${result.slug}#${createSlug(item)}`,
             category: result.category,
             wrapInCode: item.startsWith('`'),
+            locale: result.locale,
           }));
         }
 
         return result;
       };
 
-      return currentResults.slice(0, 20).map(mapResult).flat().slice(0, 20);
+      const localeResults = currentResults.filter(
+        item => item.locale === locale
+      );
+
+      const results = localeResults.length > 0 ? localeResults : currentResults;
```

Comment:
I believe this is not an appropriate solution, as `currentResults` will embed all languages. Also, filtering pre-slicing is not optimal.

Reply:
@ovflowd 

Thank you for notifying me. I just try to update the code to benchmark and put wrong logic handling. I will update it now.